### PR TITLE
HMM Performance evaluation

### PR DESCRIPTION
**Pull request by enggen**

### CODE REVIEW

**Comment by frankcholula** [comment]
we're doing isolated word recognition. I don't think insertion deletion or any of those errors are correct? We should just do the confusion matrix/ PR curve/ F1 score for a basic classification problem. Can you modify that?

File: assignment2/hmm_evaluation.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    def __init__(self, ground_truth, predicted):
+    
+        self.ground_truth = ground_truth.split() # load mfcc of original speech
+        self.predicted = predicted.split() # predicted mfcc
+        self.N = len(self.ground_truth)
+
+        # Error counts
+        self.substitution_errors = 0
+        self.deletion_errors = 0
+        self.insertion_errors = 0
```
Comment (frankcholula) [inline assignment2/hmm_evaluation.py:16]: this part doesn't quite make sense.

File: assignment2/hmm_evaluation.py [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+        print("\nConfusion Matrix:")
+        print(metrics['Confusion Matrix'])
+
+
+
+
+
+if __name__ == "__main__":
+    ground_truth = "heed hid head had hard hud hod hoard hood who'd heard"
+    predicted = "heed hid head hard hud hoard hood heard pad wood head"
```
Comment (frankcholula) [inline assignment2/hmm_evaluation.py:85]: our predicted and ground truth will be compared one file to one file. 
It's better to generate a confusion matrix of N X N to see which ones get mislabeled the most (with some sorts of heatmap)

**Review by frankcholula — CHANGES_REQUESTED** [review]

File: assignment2/hmm_evaluation.py (hunk 1 above)
Comment (enggen) [reply assignment2/hmm_evaluation.py:16]: this is store different errors but i forgot initialise

File: assignment2/hmm_evaluation.py (hunk 2 above)
Comment (enggen) [reply assignment2/hmm_evaluation.py:85]: ok i will change the metixs

**Comment by frankcholula** [comment]
closing this because eval is completed by James @sunflower-dreams-123 in #16 .

**Pull request closed without merging**